reconcile/terraform_cloudflare_dns.py

```diff
@@ -179,6 +183,21 @@ def get_desired_state_shard_config(self) -> DesiredStateShardConfig:
         )
 
 
+def are_record_identifiers_unique_within_zone(
+    zone_query_data: CloudflareDnsZoneQueryData,
+) -> bool:
+    for zone in zone_query_data.zones or []:
+        existing_records = set()
+        for record in zone.records or []:
+            record_id = record.identifier
+            if record_id not in existing_records:
+                existing_records.add(record_id)
+            else:
+                logging.warning(f"{record_id} already exists in zone {zone.identifier}")
+                return False
```

Comment:
Should we continue processing all records in case there is >1 duplicate record? If there are say 10 duplicates, it'd take several iterations through MRs to fix.

Reply:
Do you mean find all the dups in all the zones or just within that same zone? 
It's a tradeoff right, performance vs user friendliness. I don't expect MR checks to catch all the syntax mistake or other inconsistency I made at once, why is this different? Also once we make this self serviceable, it's up to the tenant to get a hint from this first error and find all the duplications.

Reply:
FYI, Gerd's suggestion about schema also don't give all the duplications, which I was hoping it did. But I still think that schema change is better.

Reply:
```
ERROR: context uniqueness error - file: /cloudflare/app-sre/dns/olivia-large-zone-1000.devshift.net.yml, path: records.[1], identifier = ['identifier'], value = test1
ERROR: context uniqueness error - file: /cloudflare/app-sre/dns/olivia-large-zone-1000.devshift.net.yml, path: records.[3], identifier = ['identifier'], value = test3
make: *** [bundle] Error 1
```